### PR TITLE
Add scheduler e2e tests

### PR DESCRIPTION
**Pull request by mvladev**
**What this PR does / why we need it**:

Adds scheduler e2e tests

**Which issue(s) this PR fixes**:
Related to #1610 #1551 

**Special notes for your reviewer**:

n/a

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: .test-defs/ShootControlPlaneTest.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+kind: TestDefinition
+metadata:
+  name: shoot-controlplane-test
+spec:
+  owner: gardener-oq@listserv.sap.com
+  description: Tests for various control plane components of the Shoot cluster.
+
+  activeDeadlineSeconds: 1800
+  labels: ["default", "release", "shoot"]
```
Comment (schrodit) [inline .test-defs/ShootControlPlaneTest.yaml]: please do not label untestet tests as `release`.
We first want to see if the test is stable enough to be a release blocker.

File: .test-defs/ShootControlPlaneTest.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+kind: TestDefinition
+metadata:
+  name: shoot-controlplane-test
+spec:
+  owner: gardener-oq@listserv.sap.com
```
Comment (schrodit) [inline .test-defs/ShootControlPlaneTest.yaml]: Please set yourself or a more specialized team as owner

File: .test-defs/ShootControlPlaneTest.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  description: Tests for various control plane components of the Shoot cluster.
+
+  activeDeadlineSeconds: 1800
```
Comment (schrodit) [inline .test-defs/ShootControlPlaneTest.yaml]: 30 minutes seem a little long for test.
I think 10 or 5 minutes is enough?

**Review by schrodit — CHANGES_REQUESTED** [review]
just some testdef related things.
The test is lgtm

**Comment by mvladev** [comment]
@schrodit updated the test defs. Have another look

**Comment by vlerenc** [comment]
Hmm... I skimmed through the code, but what is it testing? I don't see much, the description is not helping, neither for the PR nor in the test def.

**Comment by ialidzhikov** [comment]
You do only get the kube-scheduler configMap and verify its content. Can't we do this with simple unit test? Why we need integration test for it?

**Review by ialidzhikov — CHANGES_REQUESTED** [review]
As a said I don't see much added value in this test. It won't catch issues as the ones #1610. And if it even fails for some reason, then probably the kube-scheduler will be down which will block the Shoot creation.
@mvladev wdyt? Can we rework/close this PR?

**Comment by rfranzke** [comment]
> You do only get the kube-scheduler configMap and verify its content. Can't we do this with simple unit test? Why we need integration test for it?

Good point, a unit test for the [`DeployKubeScheduler`](https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/controlplane.go#L1000) function would be good enough, I think.

> It won't catch issues as the ones #1610. 

Why are you saying that this wouldn't have covered #1610? Actually, it would, right?

> And if it even fails for some reason, then probably the kube-scheduler will be down which will block the Shoot creation.

WDYM here?

**Comment by ialidzhikov** [comment]
> Why are you saying that this wouldn't have covered #1610? Actually, it would, right?

You are just verifying configMap content. You don't verify any scheduling behaviour actually. After 2 days I can go and put my algorithm `foo` and adapt the test to check for it.

**Comment by rfranzke** [comment]
Yes, but it would have covered the problem that was introduced. True point regarding actually testing the scheduling behavior, hence, let’s make it a unit test instead of an integration test.
I don’t understand the second sentence. Of course, simply adapting the test must be carefully reviewed and only be allowed if it makes sense. That’s always true, for every test we have.

**Comment by ialidzhikov** [comment]
> WDYM here?

Do you have a case when this test will fail and the Shoot creation won't be failed?

**Comment by rfranzke** [comment]
I don’t understand the discussion. Let’s make it a unit test and then we should be good to go. It’s valuable in the sense that it would catch the problem introduced last week. Yes, it’s not perfect as it doesn’t test actual scheduling behavior, though, I consider it valuable. It’s better than nothing ;-)

**Comment by ialidzhikov** [comment]
> Let’s make it a unit test and then we should be good to go

+1

**Comment by mvladev** [comment]
I've made it a e2e test to validate that the actual content of the `ConfigMap` is correct. A `MutatingWebhookConfiguration` could be present and modify the content - which is something that we don't want.

**Comment by ialidzhikov** [comment]
> I've made it a e2e test to validate that the actual content of the `ConfigMap` is correct. A `MutatingWebhookConfiguration` could be present and modify the content - which is something that we don't want.

wtf, I don't think it a valid argument. Then lets write a bunch of tests that validates the content of kube-proxy configMap, the kube-apiserver configMap, etc.

**Comment by rfranzke** [comment]
How do we proceed here? @mvladev will you adapt to a unit test as part of this PR or should we close it in favour of a new one?

**Comment by rfranzke** [comment]
Closing for now, please re-open once adapted as discussed.

**Pull request closed without merging**